core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,312 @@
+            }
+        }
+    }
+
+    /**
+     * Avoid some more ways than for pedestrian like hiking trails.
+     */
+    @Override
+    public EncodingManager.Access getAccess(ReaderWay way) {
+        if (way.hasTag("wheelchair", intendedValues)) {
+            return EncodingManager.Access.WAY;
+        }
+
+        if (way.getTag("sac_scale") != null) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("highway", avoidHighwayTags) && !way.hasTag("sidewalk", sidewalkValues)) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("surface", excludeSurfaces)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":surface", excludeSurfaces)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        if (way.hasTag("smoothness", excludeSmoothness)) {
+            if (!way.hasTag("sidewalk", sidewalkValues)) {
+                return EncodingManager.Access.CAN_SKIP;
+            } else {
+                String sidewalk = way.getTag("sidewalk");
+                if (way.hasTag("sidewalk:" + sidewalk + ":smoothness", excludeSmoothness)) {
+                    return EncodingManager.Access.CAN_SKIP;
+                }
+            }
+        }
+
+        if (way.hasTag("incline")) {
+            String tagValue = way.getTag("incline");
+            if (tagValue.endsWith("%") || tagValue.endsWith("°")) {
+                try {
+                    double incline = Double.parseDouble(tagValue.substring(0, tagValue.length() - 1));
+                    if (tagValue.endsWith("°")) {
+                        incline = Math.tan(incline * Math.PI / 180) * 100;
+                    }
+
+                    if (-maxInclinePercent > incline || incline > maxInclinePercent) {
+                        return EncodingManager.Access.CAN_SKIP;
+                    }
+                } catch (NumberFormatException ex) { }
+            }
+        }
+
+        if (way.hasTag("kerb", "raised")) {
+            return EncodingManager.Access.CAN_SKIP;
+        }
+
+        if (way.hasTag("kerb")) {
+            String tagValue = way.getTag("kerb");
+            if (tagValue.endsWith("cm") || tagValue.endsWith("mm")) {
+                try {
+                    float kerbHeight = Float.parseFloat(tagValue.substring(0, tagValue.length() - 2));
+                    if (tagValue.endsWith("mm")) {
+                        kerbHeight /= 100;
+                    }
+
+                    if (kerbHeight > maxKerbHeightCm) {
+                        return EncodingManager.Access.CAN_SKIP;
+                    }
+                } catch (NumberFormatException ex) { }
+            }
+        }
+
+        return super.getAccess(way);
+    }
```

Comment:
Can you list what you changed? Maybe we can make the super method a bit more generic and remove it here.

Reply:
Have fixed a few things. Still, can you list what you changed?

Reply:
> Still, can you list what you changed?

Will do this in the next days when I have some more time for this (hopefully).

Reply:
Regarding the `getAccess` method, what I wrote weren't changes in the sense that behavior of the super method is overwritten. My intention was to extend the restrictions. For example to skip ways with bad surface. Further restrictions for pedestrians from the super method should be applied as well as, at least in Germany, wheelchairs are legally equivalent to pedestrians and therefore must use footways. I don't know it for sure, but I can imagine, that is similar in other countries.

Overall my additions in the `getAccess` methods are:
- allow access for ways with `wheelchair=<intendedValue>`
- skip ways with `sac_scale` tag - these are hiking trails
- skip ways with `highway=<excludeHighway>` and no sidewalk tag
- skip ways with `surface=<excludeSurface>` and no sidewalk
  - same for smoothness
- skip ways with in/decline is too steep
- skip ways with high kerbs

Reply:
Thanks!

Reply:
> at least in Germany, wheelchairs are legally equivalent to pedestrians and therefore must use footways. 

That's not entirely true. I have a power chair with lights and indicators and thus am allowed to go on the road. Furthermore, going on the sidewalk is sometimes simply not possible because of high kerbs or bad surfaces.